handsontable/test/e2e/settings/fixedColumnsStart.spec.js

```diff
@@ -0,0 +1,452 @@
+          width: 200,
+          height: 200,
+          colHeaders: true,
+        });
+
+        updateSettings({
+          fixedColumnsLeft: 2
+        });
+
+        setTimeout(() => {
+          hot.scrollViewportTo(30, 30);
+        }, 100);
+
+        setTimeout(() => {
+          expect(spy.test.calls.count()).toBe(0);
+
+          done();
+          window.onerror = prevError;
+        }, 200);
+      });
+
+      it('should synchronize scroll with master table', async() => {
+        handsontable({
+          data: Handsontable.helper.createSpreadsheetData(50, 50),
+          width: 200,
+          height: 200,
+          rowHeaders: true,
+          fixedColumnsLeft: 2,
+        });
+
+        getMaster().find('.wtHolder').scrollTop(100);
+
+        await sleep(10);
+
+        expect(getLeftClone().find('.wtHolder').scrollTop()).toBe(getMaster().find('.wtHolder').scrollTop());
+      });
+    });
+
+    it('should limit fixed columns to dataset columns length', () => {
+      handsontable({
+        data: Handsontable.helper.createSpreadsheetData(3, 3),
+        fixedColumnsLeft: 3
+      });
+
+      expect(getLeftClone().find('tbody tr:eq(0) td').length).toBe(3);
+
+      updateSettings({
+        data: Handsontable.helper.createSpreadsheetData(3, 2),
+      });
+
+      expect(getLeftClone().find('tbody tr:eq(0) td').length).toBe(2);
+
+      updateSettings({
+        data: Handsontable.helper.createSpreadsheetData(3, 1),
+      });
+
+      expect(getLeftClone().find('tbody tr:eq(0) td').length).toBe(1);
+
+      updateSettings({
+        data: Handsontable.helper.createSpreadsheetData(3, 0),
+      });
+
+      expect(getLeftClone().find('tbody tr:eq(0) td').length).toBe(0);
+
+      updateSettings({
+        data: Handsontable.helper.createSpreadsheetData(3, 1),
+      });
+
+      expect(getLeftClone().find('tbody tr:eq(0) td').length).toBe(1);
+
+      updateSettings({
+        data: Handsontable.helper.createSpreadsheetData(3, 2),
+      });
+
+      expect(getLeftClone().find('tbody tr:eq(0) td').length).toBe(2);
+
+      updateSettings({
+        data: Handsontable.helper.createSpreadsheetData(3, 3),
+      });
+
+      expect(getLeftClone().find('tbody tr:eq(0) td').length).toBe(3);
+
+      updateSettings({
+        data: Handsontable.helper.createSpreadsheetData(3, 4),
+      });
+
+      expect(getLeftClone().find('tbody tr:eq(0) td').length).toBe(3);
+    });
+  });
+  describe('fixedColumnsLeft with RTL', () => {
+    const id = 'testContainer';
+
+    beforeEach(function() {
+      this.$container = $(`<div dir="rtl" id="${id}"></div>`).appendTo('body');
+    });
+
+    afterEach(function() {
+      if (this.$container) {
+        destroy();
+        this.$container.remove();
+      }
+    });
+
+    it('constructor should throw Error', () => { // todo causes another tests to crash
+      expect(() => handsontable({
+        fixedColumnsLeft: 3
+      })).toThrow();
```

Comment:
```suggestion
      })).toThrowError('The `fixedColumnsLeft` is not supported for RTL. Please use option `fixedColumnsStart`.');
```
The suggestion refines the test. This expectation might not catch the general error (type error etc.) that could be thrown in Handsontable.